@xen-orchestra/lite/src/components/infra/InfraHostItem.vue

```diff
@@ -58,11 +62,20 @@ const isCurrentHost = computed(
   () => props.hostOpaqueRef === uiStore.currentHostOpaqueRef
 );
 const [isExpanded, toggle] = useToggle(true);
+
+const { recordsByHostRef } = useVmCollection();
+
+const vmCount = computed(() => {
+  const vms = recordsByHostRef.value.get(props.hostOpaqueRef);
+  return vms?.length;
+});
```

Comment:
You also can inline it.
```suggestion
const vmCount = computed(
  () => recordsByHostRef.value.get(props.hostOpaqueRef)?.length ?? 0
);
```